### PR TITLE
fixes handling of multiple worktrees

### PR DESCRIPTION
**Pull request by MarcelKoch**
This PR allows using gingko with multiple worktrees created by `git worktree add`.

The used command is available at least with git 2.26, but I'm not sure about the minimal required version

### CODE REVIEW

**Review by upsj — APPROVED** [review]
LGTM! The feature was added in Git 2.13, it seems https://github.com/git/git/commit/a2f5a8762693d5af6dbbca714d882a21d7ba0b75

**Review by pratikvn — APPROVED** [review]
LGTM!

**Comment by upsj** [comment]
@MarcelKoch oh and another thing we may want to consider, if you want to look at it, is installing the GCF script itself into the .git folder, otherwise we will always run into issues when deleting the build folder in a worktree.

**Comment by MarcelKoch** [comment]
@upsj the GCF script is now installed into .git/hooks. I could also add a check to skip installing, if it is already there, but I'm not sure if that would behave well with updates to the script.

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,24 +16,30 @@ set(GCF_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/git-cmake-format.py)
...
 
-execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --show-toplevel
+execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --absolute-git-dir
   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
   OUTPUT_VARIABLE GCF_GIT_ROOT
   OUTPUT_STRIP_TRAILING_WHITESPACE)
 if(NOT GCF_GIT_ROOT)
   message(WARNING "Not in a git repository")
 else()
+  set(GCF_SCRIPT_GIT_DIR ${GCF_GIT_ROOT}/hooks/git-cmake-format.py)
```
Comment (yhmtsai) [inline CMakeLists.txt]: wrong name?

Comment (MarcelKoch) [reply CMakeLists.txt]: Yes, it is not ideal, perhaps `GCF_SCRIPT_TARGET_DIR`?

Comment (yhmtsai) [reply CMakeLists.txt]: maybe GCF_GIT_DIR_SCRIPT, GCF_GIT_SCRIPT or something else?
I use the last term to identify the string type. I do not know it makes sense or not

Comment (MarcelKoch) [reply CMakeLists.txt]: Your first one works fine, I will update it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -16,24 +16,30 @@ set(GCF_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/git-cmake-format.py)
...
 
   add_custom_target(format
-    ${PYTHON_EXECUTABLE} ${GCF_SCRIPT}
+    ${PYTHON_EXECUTABLE} ${GCF_GIT_DIR_SCRIPT}
     --cmake ${GIT_EXECUTABLE}
     ${ClangFormat_EXECUTABLE} -style=${GCF_CLANGFORMAT_STYLE}
     -ignore=${GCF_IGNORE_LIST}
     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
   unset(GCF_SCRIPT)
+  unset(GCF_SCRIPT_GIT_DIR)
```
Comment (yhmtsai) [inline CMakeLists.txt]:
```suggestion
  unset(GCF_GIT_DIR_SCRIPT)
```

**Review by yhmtsai — APPROVED** [review]
LGTM, the last line misses the change

**Review by upsj — CHANGES_REQUESTED** [review]
The setup doesn't seem to work yet, you probably need to update git-pre-commit-hook as well, since this uses GCF_SCRIPT to locate the formatting script. You can check by deleting the build directory and trying to commit something.

**Comment by MarcelKoch** [comment]
@upsj the problem should be solved, although I couldn't reproduce it. Also, shouldn't that error appear in earlier versions?

**Comment by upsj** [comment]
Yes, it should. Let me try again

**Review by upsj — APPROVED** [review]
Works now, thanks!
